clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -1410,7 +1410,7 @@ void tools::linkSanitizerRuntimeDeps(const ToolChain &TC,
   // libresolv.a, even if exists, is an empty archive to satisfy POSIX -lresolv
   // requirement.
   if (TC.getTriple().isOSLinux() && !TC.getTriple().isAndroid() &&
-      !TC.getTriple().isMusl())
+      !TC.getTriple().isMusl() && TC.getSanitizerArgs(Args).needsMsanRt())
```

Comment:
This is not very efficient, because `getSanitizerArgs` actually extracts them from `Args` every time it is called. I've started working on a change to fix that.

Reply:
I propose do not change this part at all, or in a separate PR

Reply:
Done in #119442.